client/src/components/exercise/lab8/components/ChatMessage.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from "react";
+import "../../../../assets/stylesheets/components/Witch.css";
+import PropTypes from "prop-types";
+import createAvatarData from "../../../body/lab/GridImages/createAvatarData";
+import Avatar from "avataaars";
+
+const ChatMessage = ({
+  username,
+  message,
+  ai_polarity,
+  useModeration,
+  onModeration,
+}) => {
+  const [userAvatar, setUserAvatar] = useState(null);
+
+  //used to remove the moderation section so this component is more reusable
+  const [displayModerationSection] = useState(useModeration || false);
+
+  //used to remove the entire message if removed is clicked
+  const [enableMessageDisplay, setEnableMessageDisplay] = useState(true);
+
+  const [disabled, setDisabled] = useState(false);
+
+  // button colors can be set to tailwind classes (interpolated into className)
+  const [keepButtonColor, setKeepButtonColor] = useState("tw-bg-[#1ec029]");
+  const [removeButtonColor, setRemoveButtonColor] = useState("tw-bg-[#c01e1e]");
+
+  function handleKeep() {
+    setDisabled(true);
+    setKeepButtonColor("tw-bg-[#1ec029] tw-opacity-80");
+    setRemoveButtonColor("tw-bg-gray-300");
+    onModeration?.();
+  }
+
+  function handleRemove() {
+    setEnableMessageDisplay(false);
+    setDisabled(true);
+    setKeepButtonColor("tw-bg-gray-300");
+    setRemoveButtonColor("tw-bg-[#c01e1e] tw-opacity-80");
+    onModeration?.();
+  }
+
+  useEffect(() => {
+    setUserAvatar(createAvatarData(1)[0]);
```

Comment:
rather than creating the avatar data and creating the component every time, can we create a function to just create a avatar component?